### PR TITLE
Autocomplete logic change

### PR DESCRIPTION
**Pull request by eugenesk24**
### Description
- Fix lint
- Pass `getSuggestions` and `onItemSelect` as props that provide different logic for autocomplete component
- Write new autocomplete logic
    - Takes in base query
    - Parses base query
    - Only suggests after source, starting from pipeCommands
    - Completed for where and dedup, needs more work for rest of commands

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/common/search/autocomplete_logic.ts [hunk 1]
```diff
@@ -335,4 +354,85 @@ export const getSuggestions = async (
...
+    const next = parseForNextSuggestion(lastCommand);
+    switch (next) {
+      case PIPE_AFTER_FIELD:
+        return fillSuggestions(currQuery, lastWord, [
+          { label: '|' },
+          { label: 'keepempty=true' },
+          { label: 'consecutive=true' },
+        ]);
+      case CLOSE_AFTER_DATA:
+        console.log('close after data');
```
Comment (mengweieric) [inline dashboards-observability/public/components/common/search/autocomplete_logic.ts:402]: Minor: do we need logging at here and other branches?

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 2]
```diff
@@ -884,7 +890,7 @@ export const Explorer = ({
     <div className="dscAppContainer">
       <Search
         key="search-component"
-        query={query[RAW_QUERY]}
+        query={tabId === 'application-analytics-tab' ? baseQuery : query[RAW_QUERY]}
```
Comment (mengweieric) [inline dashboards-observability/public/components/explorer/explorer.tsx:893]: Minor: maybe use a constant for application-analytics-tab? I saw there's another line reference the same tab name couple of lines after.

**Review by mengweieric — APPROVED** [review]

File: dashboards-observability/public/components/common/search/autocomplete_logic.ts (hunk 1 above)
Comment (eugenesk24) [reply dashboards-observability/public/components/common/search/autocomplete_logic.ts:402]: Oh I forgot to remove these! Good catch.

File: dashboards-observability/public/components/explorer/explorer.tsx (hunk 2 above)
Comment (eugenesk24) [reply dashboards-observability/public/components/explorer/explorer.tsx:893]: Yea that make sense!

**Review by mengweieric — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

**Pull request merged**